### PR TITLE
Rootfs sz

### PR DESCRIPTION
**Pull request by NathanTP**
This adds the 'rootfs-size' option to workloads that allows you to explicitly size your images. Since you can now set it to whatever you need, the default is to resize images to have 256MiB of free space. This is a massive reduction in disk usage for FireMarshal (all buildroot images go from 2GiB to ~300MiB).

Fedora builds require a more recent version of e2fsprogs than comes with centos7. I had to manually install a more recent version (built from source). Before bumping Chipyard/FireSim, we'll need to modify the machine-launch-script.

Addresses #35 

@abejgonzalez

### CODE REVIEW

**Comment by abejgonzalez** [comment]
Do you have the changes to the `machine-launch-script` somewhere? I can go ahead and try this and verify it works if the extra install is quick/easy

**Comment by NathanTP** [comment]
it should work without changes to machine-launch-script so long as you don't use fedora. The new version of e2fsprogs is only needed for Fedora. I don't have that change anywhere yet.

File: docs/source/config.rst [hunk 1]
```diff
@@ -226,6 +226,16 @@ should be absolute with respect to the workload rootfs. Files will be placed
 together in the output directory. You cannot specify the directory structure of
 the output.
 
+rootfs-size
+^^^^^^^^^^^^^^^^^
+The desired rootfs size (in human-readable units, e.g. "4GB"). This number must
```
Comment (a0u) [inline docs/source/config.rst]: The docs imply decimal units ("GB"), but `humanfriendly.format_size()` is called with `binary=True`.  The test JSON uses IEC binary units ("GiB").  Maybe this should be clarified?

Comment (NathanTP) [reply docs/source/config.rst]: I can make the example and the docs consistent, but it actually doesn't matter. humanfriendly .parse_size reads things correctly (GiB will be binary and GB will be decimal). It always prints in binary units and the suffix will indicate that (format_size is only for logging purposes). Users are free to do whatever they please.

**Pull request merged**